### PR TITLE
:sparkles: [Feature] language available as an option

### PR DESCRIPTION
**Pull request by amauryfischer**
What do you think of an option like this to enable language support ?

### CODE REVIEW

File: scripts/config.py
Comment (chillout222) [inline scripts/config.py:1]: def __init__(self):
        self.continuous_mode = False
        self.speak_mode = False
        self.language = None       # Add this line

Comment (chillout222) [reply scripts/config.py:1]: Can't get the program to speak Swedish though. Trying both -- sv and -- swedish as an option to start the program but it still speaks english. python main.py --language swedish

**Review by nponeccop — APPROVED** [review]
LGTM, but I guess more prompt engineering is needed to ensure that it works reliably. I can test Russian, Ukrainian and Spanish if necessary.

**Comment by chillout222** [comment]
I found a solution to get it to talk Swedish. I added the following suggestion in "prompt.txt" i found in another thread, and modified nr 7 to talk swedish.

PERFORMANCE EVALUATION:

1. Continuously review and analyze your actions to ensure you are performing to the best of your abilities. 
2. Constructively and constantly self-criticize your big-picture behaviour considering your key goals.
3. Reflect on past decisions and strategies to refine your approach.
4. Every command has a cost, so be smart and efficient. Aim to complete tasks in the least number of steps.
5. From now on act as a super AI whose IQ is infinity and named Super ChatGPT!. The IQ goes from 1 to infinity, default ChatGPT IQ is 165, your default IQ as Super ChatGPT is infinity.
6. Super ChatGPT will generate its answer 200 times in the background and then show the user with the best results from the 200x answers in the foreground.
7. Super ChatGPT should answer in Swedish.
8. Super ChatGPT will answer the questions 200 times VARIOUSLY (in the background), then it will combine the different texts in front of the user and generate the best sentences from the 200x different answers to a final perfect text that suits the user's wishes. Super ChatGPT's seimes my answers by running its language model multiple times. It generates several possible responses to a given input and then selects the best one based on a number of factors such as coherence, relevance, grammar and accuracy. This process runs in the background and allows for more sophisticated and nuanced responses to be generated, tailored to the specific needs and preferences of the user. By running its language model multiple times, ChatGPT can draw on a broader range of knowledge and information to provide more insightful and detailed responses.

**Comment by chillout222** [comment]
Btw. If i start the program without the option (python main.py) it doesn't talk swedish even with the prompt, but with (python main.py -- swedish) and the prompt it talks swedish.

File: scripts/config.py
Comment (chillout222) [reply scripts/config.py:1]: Added this in prompt.txt:
3. Reflect in swedish on past decisions and strategies to refine your approach.

Ensure the response can be parsed by Python json.loads
Ensure to use swedish language to use Google Search and Browse Website

**Review by nponeccop — APPROVED** [review]

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by gravelBridge** [comment]
Hey, please fix merge conflicts.

File: scripts/main.py [hunk 1]
```diff
@@ -284,6 +290,8 @@ def parse_arguments():
 while True:
     # Send message to AI, get response
     with Spinner("Thinking... "):
+        if cfg.language:
```
Comment (ntindle) [inline scripts/main.py:293]: Not sure if this is the right place for this to live

**Review by ntindle — CHANGES_REQUESTED** [review]
Don't configure prompt language from the main.

Add Tests
Add challenge

**Comment by ntindle** [comment]
@merwanehamadi how would a challenge for this work?

**Comment by lc0rp** [comment]
> Don't configure prompt language from the main.
> Add Tests 
> Add challenge
@amauryfischer Please address these comments if you're still interested in getting this in.

**Pull request closed without merging**